api/common/sharing/utils.go

```diff
@@ -135,11 +132,47 @@ func getSharedInstances(ctx context.Context, repository storage.Repository, offe
 	return objectList, err
 }
 
-func filterInstancesBySelectors(ctx context.Context, repository storage.Repository, instances types.ObjectList, parameters map[string]gjson.Result, smaap bool, offeringID string) (*types.ServiceInstance, error) {
+func filterInstancesBySelectors(ctx context.Context, repository storage.Repository, offeringID, tenantIdentifier, tenantValue string, selectors map[string]gjson.Result, smaap bool) (*types.ServiceInstance, error) {
 	var err error
 	var filteredInstances types.ServiceInstances
 
-	planSelectorVal, planSelectorExists := parameters[instance_sharing.ReferencePlanNameSelectorKey]
+	var sharedInstances types.ObjectList
+
+	// Filter by label query:
+	labelsSelector, labelExists := selectors[instance_sharing.ReferenceLabelSelectorKey]
+	labelsSelectorArray := labelsSelector.Array()
```

Comment:
labelsSelector.Array() - what happens if the value doesn't exist ? should not you  check  labelExists before doing labelSelector.Array()

Reply:
fixing